api/src/main/java/org/ehcache/spi/service/PersistableResourceService.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.service;
+
+import org.ehcache.config.ResourcePool;
+import org.ehcache.config.ResourceType;
+import org.ehcache.exceptions.CachePersistenceException;
+
+import java.util.Collection;
+
+/**
+ * PersistableResourceService
+ */
+public interface PersistableResourceService extends Service {
+
+  boolean handlesResourceType(ResourceType resourceType);
```

Comment:
This looks similar to what Clifford and I were discussing today... except that we had started extending these `PersistableResourceService` instances to have two additional properties:

1. They create stores (which seemed to indicate that this method would consume Set<ResourceType> rather than just a single resource)
2. They have state machines of their own... thereby decoupling the maintainable state of the local cache manager from the maintainable state of any one of a number of potential maintainable states of associated persistence services.